judo-runtime-core-jsl-itest/models/CompositionRelationshipsModel/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/CompositionRelationshipsTest.java

```diff
@@ -364,4 +394,87 @@ void testDeepCopyConstructor() {
+        // TODO: JNG-4194
+        EntityF f = EntityF.builder().build();
+
+        // Default attribute value is undefined on required attribute: _name_Default_EntityG
+        //f.setG(EntityG.builder().withName("Entity").build());
+        //f = entityFDao.create(f);
+        //assertTrue(f.getG().orElseThrow().getName().equals("Entity"));
+
+        EntityF f12 = EntityF.builder().build();
+
+        EntityH h = entityHDao.create(EntityH.builder().withAlwaysUndefined("Entity").build());
+        f12.setG(EntityG.builder().build());
+        f12 = entityFDao.create(f12);
+
+        assertTrue(f12.getG().orElseThrow().getName().equals("Entity"));
+
+        entityHDao.delete(h);
+
+        EntityF f13 = entityFDao.create(EntityF.builder().build());
+        // Default attribute value is undefined on required attribute: _name_Default_EntityG
+        //EntityG g13 = entityFDao.createG(f13, EntityG.builder().withName("Entity").build());
+        //assertTrue(g13.getName().equals("Entity"));
+        //assertTrue(f13.getG().orElseThrow().getName().equals("Entity"));
+
+        EntityF2 f2 = EntityF2.builder().build();
+
+        f2.setH(EntityH.builder().withAlwaysUndefined("Entity").build());
+        f2 = entityF2Dao.create(f2);
+
+        assertTrue(f2.getH().orElseThrow().getAlwaysUndefined().orElseThrow().equals("Entity"));
+        entityHDao.delete(f2.getH().orElseThrow());
+
+        // java.lang.NullPointerException
+        EntityF2 f21 = entityF2Dao.create(EntityF2.builder().build());
+        EntityH h2 = entityF2Dao.createH(f21, EntityH.builder().withAlwaysUndefined("Entity").build());
+        //java.util.NoSuchElementException: No value present
+        //assertTrue(f21.getH().orElseThrow().getAlwaysUndefined().orElseThrow().equals("Entity"));
+        //java.lang.NullPointerException
+        //assertTrue(h2.getAlwaysUndefined().orElseThrow().equals("Entity"));
+
+        entityHDao.delete(h2);
+
+        EntityF3 f3 = EntityF3.builder().build();
+
+        f3.setG(List.of(EntityG.builder().withName("Entity2").build()));
+        //Default attribute value is undefined on required attribute: _name_Default_EntityG
+        //f3 = entityF3Dao.create(f3);
+//
+        //assertTrue(f3.getG().get(0).getName().equals("Entity2"));
+
+        h2 = entityF2Dao.createH(f21, EntityH.builder().withAlwaysUndefined("Entity").build());
+
+        EntityF3 f31 = EntityF3.builder().build();
+
+        f31.setG(List.of(EntityG.builder().build()));
+        f31 = entityF3Dao.create(f31);
+
+        assertTrue(f31.getG().get(0).getName().equals("Entity"));
+
+        entityHDao.delete(h2);
+
+        EntityF3 f32 = entityF3Dao.create(EntityF3.builder().build());
+        // Default attribute value is undefined on required attribute: _name_Default_EntityG
+        //EntityG g3 = entityF3Dao.createG(f32, EntityG.builder().withName("Entity2").build());
+        //assertTrue(g3.getName().equals("Entity2"));
```

Comment:
Add the bug ticket number for composition create method dosnt get back the instance

Reply:
[60a12f2](https://github.com/BlackBeltTechnology/judo-runtime-core-jsl/pull/223/commits/60a12f22a8ad517f643d5d01280a3959bee7af23)